Sources/SignalProducer.swift

```diff
@@ -65,6 +65,23 @@ public struct SignalProducer<Value, Error: Swift.Error> {
 		}
 	}
 
+	/// Creates a producer for a `Signal` that immediately sends one value, then
+	/// completes.
+	///
+	/// This initializer differs from `init(value:)` in that it will not 
+	/// evaluate the closure supplying its sole `value` event until the 
+	/// `SignalProducer` is started.
+	///
+	/// - parameters:
+	///   - block: A block that supplies a value to be sent by the `Signal` in
+	///            a `value` event.
+	public init(_ block: @escaping () -> Value) {
```

Comment:
@NachoSoto Interesting. I wonder, though, whether we'd still have an issue where the compiler wants to forcefully optimize a passed-in value as being evaluated immediately.

For example, consider the case of `SignalProducer(value: callAFunction())` versus the `SignalProducer({ callAFunction() })` construct. Would the former be turned into an immediately evaluated variant if the braces are omitted, versus being evaluated lazily? 

That seems a little subtle/delicate from a usage standpoint… But yeah—having the `lazyValue:` as I originally built it might be a good tiebreaker?